src/plugins/MamicoCoupling.cpp

```diff
@@ -0,0 +1,80 @@
+#include "MamicoCoupling.h"
+#include "Domain.h"
+
+#ifdef MAMICO_COUPLING
+#include <coupling/interface/impl/ls1/LS1MamicoCouplingSwitch.h>
+//#include <LS1MamicoCouplingSwitch.h>
+#endif
+
+void MamicoCoupling::readXML(XMLfileUnits& xmlconfig)
+{	
+    return;
+}
+
+void MamicoCoupling::init(ParticleContainer* particleContainer,
+		DomainDecompBase* domainDecomp, Domain* domain)
+{
+    #ifdef MAMICO_COUPLING
+    if(_macroscopicCellService==NULL) 
+    {
+        _macroscopicCellService = (coupling::services::MacroscopicCellServiceImpl<ls1::LS1RegionWrapper,3>*)
+                coupling::interface::MamicoInterfaceProvider<ls1::LS1RegionWrapper,3>::getInstance().getMacroscopicCellService();
+	}
+    //since using mamico thermostat, switch off ls1 thermostat
+    domain->thermostatOff();
+	//code to print to log that plugin is initialised
+    #endif
+}
+
+void MamicoCoupling::beforeEventNewTimestep(ParticleContainer* particleContainer,
+		DomainDecompBase* domainDecomp, unsigned long simstep)
+{
+	
+}
+
+void MamicoCoupling::beforeForces(ParticleContainer* particleContainer,
+		DomainDecompBase* domainDecomp, unsigned long simstep)        
+{
+    #ifdef MAMICO_COUPLING
+	if(coupling::interface::LS1MamicoCouplingSwitch::getInstance().getCouplingState())
+    {
+        if(_macroscopicCellService==NULL) 
```

Comment:
same as above
```suggestion
        if(not _macroscopicCellService) 
```

Reply:
Why do we have to check this in every plugin step?

Reply:
This object is often switched around by mamico esp. when multiple MD instances get involved, so it acts as a failsafe (maybe not strictly necessary, but still)

Reply:
Ok, add a comment to the code explaining this.